test/req/steps_test.exs

```diff
@@ -320,6 +320,16 @@ defmodule Req.StepsTest do
     assert Req.get!(c.url).body == %{"a" => 1}
   end
 
+  test "decode_body/1: json-api", c do
+    Bypass.expect(c.bypass, "GET", "/", fn conn ->
+      conn
+      |> Plug.Conn.put_req_header("content-type", "application/vnd.api+json; charset=utf-8")
+      |> json(200, %{a: 1})
```

Comment:
this test passes on the main branch already but for the wrong reason. It is because the `json/3` function incorrectly set the `application/json` content-type whereas it should only do that if content-type is not set on the request. Could you fix that one while at it too?

Reply:
updated `json/3` to not override the content-type, also confirmed this test breaks in main, thanks!